From c510593b5ee417066f71b0e00e52c17ad325122a Mon Sep 17 00:00:00 2001
From: Thiruvel Thirumoolan <thiruvel@yahoo-inc.com>
Date: Fri, 20 Dec 2013 14:09:00 -0800
Subject: [PATCH 0377/1164] CDH-31318: HIVE-6091 : Empty pipeout files are
 created for connection create/close (Thiruvel
 Thirumoolan, Bing Li via Ashutosh Chauhan)

Change-Id: I50196c00201ad960fb8c9aefc07684b35c83e668
Signed-off-by: Ashutosh Chauhan <hashutosh@apache.org>
---
 .../hadoop/hive/ql/session/SessionState.java       |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java b/ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java
index 21b60bd..36f9c60 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/session/SessionState.java
@@ -703,6 +703,9 @@ private void dropSessionPaths(Configuration conf) throws IOException {
     if (localSessionPath != null) {
       FileSystem.getLocal(conf).delete(localSessionPath, true);
     }
+    if (this.getTmpOutputFile().exists()) {
+      this.getTmpOutputFile().delete();
+    }
   }
 
   /**
-- 
1.7.9.5

